src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Encode.cs

```diff
@@ -28,8 +28,18 @@ internal static byte[] OpenSslEncode<THandle>(GetEncodedSizeFunc<THandle> getSiz
             byte[] data = new byte[size];
 
             int size2 = encode(handle, data);
-            Debug.Assert(size == size2);
+            if (size2 < 1)
+            {
+                Debug.Fail(
+                    $"{nameof(OpenSslEncode)}: {nameof(getSize)} succeeded ({size}) and {nameof(encode)} failed ({size2})");
 
+                // If it ever happens, ensure the error queue gets cleared.
+                // And since it didn't write the data, reporting an exception is good too.
+                throw CreateOpenSslCryptographicException();
+            }
```

Comment:
Is this necessary?  Presumably we're asserting because it's never possible.

Reply:
It is more on the paranoid side but at least for one of the encode delegates called via this method the OpenSsl docs states the following: "[This may be fixed in future so code should not assume that i2d_X509() will always succeed.](https://www.openssl.org/docs/man1.0.2/crypto/i2d_X509.html)" - granted that it is in the context of not fully initialized struct. 

@bartonjs what is your take here?

Reply:
It depends on the goals.  We've never seen the assert fire, so this is probabilisticly dead code.

Under weird not-supported multi-threading modes, though, maybe one of the values being sent to Encode is capable of failing; and that would then cause an error to show up in SslStream.

If we're going for "perfect" then this is warranted.  If we're going for "probabilistic" then it isn't.

Reply:
Let's go for "asymptotically perfect" ...